packages/babel-plugin-proposal-decorators/src/index.ts

```diff
@@ -12,18 +12,24 @@ import transformer2021_12 from "./transformer-2021-12";
 export default declare((api, options) => {
   api.assertVersion(7);
 
-  const { legacy = false } = options;
+  let { legacy = false } = options;
   if (typeof legacy !== "boolean") {
     throw new Error("'legacy' must be a boolean.");
   }
+  if (version !== undefined && options.legacy !== undefined) {
+    throw new Error(
+      'You can either specify `legacy: true` or `version: "legacy"` with decorators, not both.',
+    );
+  }
+  legacy ||= version === "legacy";
```

Comment:
When `version` is `legacy`, it should be equivalent to `legacy: true` option is set.